var/spack/repos/builtin/packages/mpt/package.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+import os
+
+class mpt(Package):
+    """HPE MPI is HPE's implementation of 
+    the Message Passing Interface (MPI) standard.
+ 
+    Note: HPE MPI is proprietry software. Spack will search your
+    current directory for the download file. Alternatively, add this file to a
+    mirror so that Spack can find it. For instructions on how to set up a
+    mirror, see http://spack.readthedocs.io/en/latest/mirrors.html"""
+
+    homepage = ""
+
+    provides('mpi')
+    provides('mpi@:3.1', when='@3:')
+    provides('mpi@:1.3', when='@1:')
+
+    filter_compiler_wrappers(
+        'mpicc', 'mpicxx', 'mpif77', 'mpif90', 'mpif77', 'mpif08', relative_root='bin'
+    )
+
+    def setup_dependent_environment(self, spack_env, run_env, dependent_spec):
+        spack_env.set('MPICC',  join_path(self.prefix.bin, 'mpicc'))
+        spack_env.set('MPICXX', join_path(self.prefix.bin, 'mpicxx'))
+        spack_env.set('MPIF77', join_path(self.prefix.bin, 'mpif77'))
+        spack_env.set('MPIF90', join_path(self.prefix.bin, 'mpif90'))
+
+        spack_env.set('MPICC_CC', spack_cc)
+        spack_env.set('MPICXX_CXX', spack_cxx)
+        spack_env.set('MPIF90_F90', spack_fc)
+
+    def setup_dependent_package(self, module, dependent_spec):
+        if 'platform=cray' in self.spec:
+            self.spec.mpicc = spack_cc
+            self.spec.mpicxx = spack_cxx
+            self.spec.mpifc = spack_fc
+            self.spec.mpif77 = spack_f77
+        else:
+            self.spec.mpicc = join_path(self.prefix.bin, 'mpicc')
+            self.spec.mpicxx = join_path(self.prefix.bin, 'mpicxx')
+            self.spec.mpifc = join_path(self.prefix.bin, 'mpif90')
+            self.spec.mpif77 = join_path(self.prefix.bin, 'mpif77')
+
+    @property
+    def fetcher(self):
+        msg = """This package is a placeholder for HPE MPI, a
+        system-provided, proprietary MPI implementation.
```

Comment:
Can you show me what the error message looks like? Trying to decide if we actually want to indent this.

Reply:
[adrianj@cirrus-login0 packages]$ spack fetch mpt
==> Warning: There is no checksum on file to fetch mpt@2.18 safely.
==>   Fetch anyway? [y/N] y
==> Fetching with no checksum.
  Add a checksum or use --no-checksum to skip this check.
==> Error: This package is a placeholder for HPE MPI, a
        system-provided, proprietary MPI implementation.

        Add to your packages.yaml (changing the /opt/ path to match
        where HPE MPI is actually installed):

        packages:
          mpt:
            paths:
              mpt@2.20: /opt/
            buildable: False